instance/models/openedx_instance.py

```diff
@@ -164,6 +163,12 @@ def get_active_appservers(self):
         """
         Returns a queryset containing the active appservers.
         """
+        if hasattr(self.ref, '_cached_active_appservers'):
+            # A database optimization like prefetch_related() hascomputed the active
+            # appservers for a large number of instances and cached that result on
+            # the InstanceReference of each one.
+            # (This is used to optimize the /api/v1/instances/ endpoint query for example)
+            return self.ref._cached_active_appservers
```

Comment:
Cache invalidation is certainly a potential issue here, but so far the only time this cache is used is when retrieving the list of appservers, which is not an operation that modifies any models. I also think this is not a big "landmine" because `_cached_active_appservers` can only ever be populated very explicitly and deliberately by assigning some result to that specific field using `prefetch_related` with extra arguments.